### PR TITLE
🐛 Fixes issue in ObservableList while using Iterables.

### PR DESCRIPTION
**Pull request by ParthBaraiya**
Describe the changes proposed in this Pull Request.

This PR Fixes issue #941 .

---
## Pull Request Checklist

- [x] If the changes are being made to code, ensure the **version in `pubspec.yaml`** is updated. 
- [x] Increment the **`major`/`minor`/`patch`/`patch-count`**, depending on the complexity of change
- [x] Add the necessary **unit tests** to ensure the coverage does not drop
- [x] Update the **Changelog** to include all changes made in this PR, organized by version
- [x] Run the **`melo run set_version` command** from the root directory
- [x] Include the **necessary reviewers** for the PR
- [ ] Update the docs if there are any API changes or additions to functionality

### CODE REVIEW

File: mobx/lib/src/api/observable_collections/observable_list.dart [hunk 1]
```diff
@@ -336,7 +341,7 @@ class ObservableList<T>
       if (iterable.isNotEmpty) {
         final oldValues = _list.sublist(index, index + iterable.length);
         final newValues = iterable.toList(growable: false);
```
Comment (pavanpodila) [inline mobx/lib/src/api/observable_collections/observable_list.dart:343]: `newValues` can be created first and then used to fetch the `length`. That way the previous call on `iterable.length` is not needed (which actually forces a complete walk of the iterable)

**Review by pavanpodila — CHANGES_REQUESTED** [review]
Also need pubspec update, changelog and running the `melos run set_version` command to update the `lib/mobx.dart` file

**Comment by ParthBaraiya** [comment]
Hi @pavanpodila , I've made necessary changes in the PR 

But while running `melos run set_version` command it was not doing anything but to open `expose_version.dart` file in Text editor. So, I had to update the `melos.yaml` file as well. Hope that's OK.

**Comment by ParthBaraiya** [comment]
Also, I was not able to use the latest version of melos. Because latest version requires melos package added in `pubspec.yaml` and root folder doesn't have `pubspec.yaml` file.

File: flutter_mobx/lib/version.dart [hunk 2]
```diff
@@ -1,4 +1,4 @@
 // Generated via set_version.dart. !!!DO NOT MODIFY BY HAND!!!
 
 /// The current version as per `pubspec.yaml`.
-const version = '2.0.6+5';
```
Comment (pavanpodila) [inline flutter_mobx/lib/version.dart:4]: looks like this version was lagging behind and got updated once the set_version was run. Could you please include a changelog entry for this by reading the changes done in the past version.

**Review by pavanpodila — CHANGES_REQUESTED** [review]
thanks for fixing the melos script. We have one change in flutter_mobx due to a lagging version update. Can you please include a small changelog for that..might require reading the git log

**Comment by pavanpodila** [comment]
one of the test has failed with this change. pls see @ParthBaraiya

**Comment by ParthBaraiya** [comment]
> thanks for fixing the melos script. We have one change in flutter_mobx due to a lagging version update. Can you please include a small changelog for that..might require reading the git log

Looks like you have already merged a changes for this on master.

**Comment by ParthBaraiya** [comment]
> one of the test has failed with this change. pls see @ParthBaraiya

This is fixed.

**Comment by pavanpodila** [comment]
@all-contributors add @ParthBaraiya for code, bug

**Pull request merged**